### ISSUE TITLE
[Suggestion] Texture rotation/randomization

### ISSUE THREAD

**Issue by chland** [llbit/chunky#435]
Quite a few years ago, Optifine added a feature called "Natural Textures". If you enabled it, it rotated the textures for certain blocks like sand or grass. The same seems nowadays be possible by using resource-packs (? sorry, I really don't know much about that stuff) and block-variants.

AFAIK, chunky doesn't really care much about block-model stuff ATM. So i guess thats the reason randomizing the textures that way doesn't work. But I was wondering if it would be possible to add a checkbox to either the "Advanced"-tab to enable random rotation for certain blocks like grass, sand, stone, snow, etc. or add an "Allow Texture Rotation" checkbox to the new material-config (probably even divided into different settings for the top/bottom/sides of the block). And if you enable that feature, chunky just rotates the texture randomly by 0/90/180/270 for every block.

Here is a little screenshot that shows the difference:

![random_texture](https://user-images.githubusercontent.com/472894/27912238-6518bf20-625c-11e7-963c-db53ac04b9e2.jpg)

**Comment by chland** [comment]
Oops: just saw that https://github.com/llbit/chunky/issues/400 is similar to my suggestion.

**Comment by syrinxsean** [comment]
It's similar in goal but very different in proposed technique to reach that
goal.

-Sean

--
Sean Ahern
Computational Engineering International
919-363-0883

**Comment by chland** [comment]
Yeah, my idea is a lot simpler and doesn't affect the grass-plant/bush/blades (whatever it is called :)) but only the texture of the block beneath it to get rid of the moiré pattern thats caused by the repeated textures.

**Comment by leMaik** [comment]
Related to #137 (which proposes to use different animation frames for animated textures)
